teku/src/main/java/tech/pegasys/teku/cli/subcommand/debug/DebugDbCommand.java

```diff
@@ -432,6 +435,244 @@ public int dumpBlobSidecars(
+                      .getBody()
+                      .getOptionalBlobKzgCommitments()
+                      .map(SszList::size)
+                      .orElse(0);
+              for (int blobSidecarIndex = 0;
+                  blobSidecarIndex < expectedBlobSidecarsCount;
+                  blobSidecarIndex++) {
+                final SlotAndBlockRootAndBlobIndex blobSidecarKey =
+                    new SlotAndBlockRootAndBlobIndex(
+                        hotBlockSlot, hotBlockRoot, UInt64.valueOf(blobSidecarIndex));
+
+                // Checking blob sidecar existence
+                final Optional<BlobSidecar> maybeCurrentBlobSidecar =
+                    database.getBlobSidecar(blobSidecarKey);
+                if (maybeCurrentBlobSidecar.isEmpty()
+                    && hotBlockSlot.isGreaterThanOrEqualTo(lowerSlotBoundDataAvailability)) {
+                  // inside DA window but missing
+                  missingHotBlobsSidecars++;
+                  System.err.printf(
+                      "ERROR: Unable to locate hot blob sidecar %s%n", blobSidecarKey);
+                }
+              }
+            }
+          }
+        }
+      }
+
+      // Checking orphaned blob sidecars
+      try (final Stream<SlotAndBlockRootAndBlobIndex> blobsSidecarKeys =
+          database.streamBlobSidecarKeys(UInt64.ZERO, UInt64.MAX_VALUE)) {
+        for (final Iterator<SlotAndBlockRootAndBlobIndex> blobSidecarsIterator =
+                blobsSidecarKeys.iterator();
+            blobSidecarsIterator.hasNext(); ) {
+          final SlotAndBlockRootAndBlobIndex blobSidecarKey = blobSidecarsIterator.next();
+
+          sidecarsToBlocksCheckCounter++;
+
+          final Optional<SignedBeaconBlock> maybeFinalizedBlockFound =
+              database
+                  .getFinalizedBlockAtSlot(blobSidecarKey.getSlot())
+                  .filter(block -> block.getRoot().equals(blobSidecarKey.getBlockRoot()));
+
+          final Optional<SignedBeaconBlock> maybeHotBlockFound =
+              database.getHotBlock(blobSidecarKey.getBlockRoot());
+
+          if (maybeHotBlockFound.isEmpty() && maybeFinalizedBlockFound.isEmpty()) {
+            orphanBlobsSidecars++;
+            System.err.printf(
+                "ERROR: blob sidecar has no corresponding finalized or hot block %s%n",
+                blobSidecarKey);
+          }
+
+          // check Data Availability
+          if (blobSidecarKey.getSlot().isLessThan(lowerSlotBoundDataAvailability)) {
+            unprunedBlobsSidecars++;
+            System.err.printf("WARNING: non pruned blobs sidecar %s%n", blobSidecarKey);
+          }
+        }
+      }
+    } catch (DatabaseStorageException ex) {
+      System.out.println("Failed to open database");
+    }
+
+    final long endTimeMillis = System.currentTimeMillis();
+    System.out.printf(
+        "Done. Performed %s block->blobs_sidecars and %s blobs_sidecars->block checks in %s ms%n",
+        blocksToSidecarsCheckCounter,
```

Comment:
I think originally it was expected that these 2 numbers match if everything is correct, now `blocksToSidecarsCheckCounter` is blocks counter and `sidecarsToBlocksCheckCounter` is sidecars counter. So I think you should increment `blocksToSidecarsCheckCounter` by the number of commitments

Reply:
mm.. ye, there is value in checking that numbers are the same, so we could print `block_commitments->blob_sidecars`

Reply:
Added a `commitmentsToBlobSidecarsCheckCounter` which counts the expected blob sidecars according to the block commitments. This way the `sidecarsToBlocksCheckCounter` and the new `commitmentsToBlobSidecarsCheckCounter` should match if everything is fine. Decided to leave the block checks counter separate but could change that if you think it's better

Reply:
But the message looks correct
```
Done. Performed 163 block->blobs_sidecars and 491 blobs_sidecars->block checks in 4401 ms
```
I'd add just another counter for commitments count and add it to the message

Reply:
@mehdi-aouadi yeeah, it's what I was going to suggest. Perfect!